### PR TITLE
ROX-22101: use gogo TimestampNow through abstraction layer

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The goal is to remove direct calls to the gogo protobuf layer and have all of them happen through an abstraction layer in order to ease the migration to a new protobuf library.

This PR moves the calls to TimestampNow from the gogo protobuf package to the protocompat abstraction layer.

The change is the application through the refactoring tool `eg` of the following template:
```
package refactor

import (
	gogoProto "github.com/gogo/protobuf/types"
	"github.com/stackrox/rox/pkg/protocompat"
)

func before() *gogoProto.Timestamp {
	return gogoProto.TimestampNow()
}

func after() *gogoProto.Timestamp {
	return protocompat.TimestampNow()
}
```

PR stack:
- #9148
  - #9156 
  - #9145 
  - #9196 
  - #9198
  - #9201

## Checklist
- [ ] Investigated and inspected CI test results
~~- [ ] Unit test and regression tests added~~
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: pkg/administration/events/event.go [hunk 1]
```diff
@@ -3,7 +3,9 @@ package events
 import (
 	"strings"
 
-	gogoTimestamp "github.com/gogo/protobuf/types"
+	// TODO: find a way to break the dependency loop when pulling pkg/protoconv
```
Comment (RTann) [inline pkg/administration/events/event.go:6]: yeah I don't think the logging library should depend on anything outside of itself and the standard library, but that's unrelated to this PR

**Review by RTann — APPROVED** [review]

**Review by janisz — APPROVED** [review]

File: pkg/administration/events/event.go (hunk 1 above)
Comment (janisz) [reply pkg/administration/events/event.go:6]: How about removing this logger from converting function?

https://github.com/stackrox/stackrox/blob/6ed966ddbe6c1c22876fdfac6de6c555445ae0ff/pkg/protoconv/time.go#L57-L61

**Comment by dashrews78** [comment]
I reviewed this right after I merged PR #9013 so it looks like I added a spot you need to update `central/convert/testutils/compliance_scans.go`

**Review by md2119 — APPROVED** [review]
I have the same comment as [this one](https://github.com/stackrox/stackrox/pull/9196#pullrequestreview-1806773059), otherwise LGTM.

**Pull request merged**